cube/src/main/java/zone/cogni/asquare/cube/convertor/data2shacl/ShaclGenerator.java

```diff
@@ -348,29 +315,42 @@ private Resource calculateNodeKind(boolean hasIri, boolean hasBlank, boolean has
     return null;
   }
 
-  private void setShaclDatatype(Configuration configuration,
-                                RdfStoreService rdfStoreService,
-                                Model shacl,
-                                Resource targetClass,
-                                Resource path,
-                                Resource propertyShape) {
-    List<String> datatypes = selectForTypeAndProperty(rdfStoreService, "select-datatype.sparql.spel",
-                                                      targetClass, path);
+  private void setShaclDatatype(@Nonnull RdfStoreService rdfStore,
+                                @Nonnull Model shacl,
+                                @Nonnull Resource targetClass,
+                                @Nonnull Resource path,
+                                @Nonnull Resource propertyShape) {
+    List<String> datatypes = selectUris(rdfStore, "select-datatype.sparql.spel", getTypeAndPropertyParameters(targetClass, path));
     if (datatypes.size() != 1) {
       String message = getMessage("type '{}' and property '{}' does not have exactly one datatype: {}",
                                   shortenUri(shacl, targetClass), shortenUri(shacl, path), shortenUri(shacl, datatypes));
       log.warn(message);
```

Comment:
as next block is gone, this can be direct call:
```suggestion
      log.warn("type '{}' and property '{}' does not have exactly one datatype: {}",
                     shortenUri(shacl, targetClass), shortenUri(shacl, path), shortenUri(shacl, datatypes));
```

Reply:
fixed via IntelliiJ